[DM-347] As user I want user docs for advanced software management

bozhidardedov opened the pull request.

rssag reviewed (approved):
language review as commit + approval

dafkk commented on content/users-guide/device-management-bundle/device-details.md:
```diff
@@ -393,15 +393,19 @@ To configure LAN parameters, simply enter **IP address** and **Subnet mask**.
 ![Status filter](/images/users-guide/DeviceManagement/devmgmt-services-status-filter.png)
 
-Other status options are also allowed and can be freely defined. They all would share the same icon and need to be manually entered in the **Status** filter. A tooltip on the status icon will display their actual value:
+Other status options are also allowed and can be freely defined.
```
I was wondering if we can add here a link to the resources docs with the way to add a new option, but I am not sure if that is published. I am talking about this here - packages/ngx-components/services/services.module.md. Does anyone knows @rssag, @BeateRixen?

rssag replied:
what would the full link be?

bozhidardedov replied:
The link to this part of the published WebSDK documentation should be http://resources.cumulocity.com/documentation/websdk/ngx-components/modules/ServicesModule.html#readme

<img width="1546" alt="image" src="https://user-images.githubusercontent.com/95280800/162421702-1e91af6a-6610-4548-9532-2b73e1ad2772.png">

rssag replied:
this yields a 404 - will the URL work at some point?

bozhidardedov replied:
Once WebSDK doc has been built for 10.14.0.x, the link should start working.

rssag replied:
added the URL

BeateRixen commented on content/users-guide/device-management-bundle/managing-device-data.md:
```diff
@@ -189,14 +189,17 @@ The available software objects will be displayed as a list.
+At the top, the software name, a description, an optional device type filter(s), and a software type are shown.
+If a device type filter is set, the software will show up for installation only for devices of that type.
+If no filter is set, it will be available for all devices.
+The software type will make the software installable only on devices that have declared to support the particular software type.
```
@rssag "only on devices that have declared to support" 
I think we can get that simpler :-). Suggestion:  "only on devices that explicitly support"? Also a bit further below.

rssag replied:
i like "that specifically support"

bozhidardedov replied:
Device MOs can have a fragment in which the software types they support are listed. Thus "only on devices that explicitly support" should make that clear, I guess.

rssag replied:
done

BeateRixen commented on content/users-guide/device-management-bundle/managing-device-data.md:
```diff
@@ -260,9 +264,19 @@ Additionally, it shows the operation status for the last operation (one of SUCCE
+
+    >**Info:** The **Install software** dialog will only display software entries matching the device type. Additionally, if the device has any `c8y_SupportedSoftwareTypes` declared the dialog will only display the software entries matching the supported software types.
+
+2. Select one or multiple softwares by selecting the respective version from the list which contains all softwares for the particular device type available in the software repository. For devices supporting advanced software management features, already installed software entries cannot be pre-filtered from the list of available software entries. Thus, after a particular software version has been ...
```
softwares > software items; several times
software entries > software item
chosen > selected

rssag replied:
done, found a couple more occurrences.

BeateRixen commented on content/users-guide/device-management-bundle/managing-device-data.md:
```diff
@@ -260,9 +264,19 @@ Additionally, it shows the operation status for the last operation (one of SUCCE
+
+    ![Already installed software entry warning](/images/users-guide/DeviceManagement/devmgmt-software-install-warning.png)
+
+    You can remove the already installed software entry from the **Software changes** panel. However, if the resulting operation includes an already installed software entry this has no effect on the device software.
```
entry > item
panel > we don´t use this; not sure what this looks like?

bozhidardedov replied:
Typo, "pane" is meant.

rssag replied:
i think we can just avoid panel and only use the UI text.

BeateRixen commented on content/users-guide/device-management-bundle/managing-device-data.md:
```diff
@@ -260,9 +264,19 @@ Additionally, it shows the operation status for the last operation (one of SUCCE
+    ![Software changes warning](/images/users-guide/DeviceManagement/devmgmt-software-changes-warning.png)
+
+3. Click **Install**.
+4. In the **Software changes** panel at the right, review your planned changes and confirm the software update operation by clicking **Apply changes**.<br><br>
```
panel?

bozhidardedov replied:
Again "pane".

rssag replied:
see above

BeateRixen commented:
@bozhidardedov There are a couple of screenshots showing tooltips or messages. I would avoid these kinds of screenshots and remove them. They are of not much value for the user. Instead, you could simply provide the warning/tooltip message in the text. Be aware that screenshot maintenance kills us (for example, if we again have a design change) :-).

bozhidardedov commented:
> @bozhidardedov There are a couple of screenshots showing tooltips or messages. I would avoid these kinds of screenshots and remove them. They are of not much value for the user. Instead, you could simply provide the warning/tooltip message in the text. Be aware that screenshot maintenance kills us (for example, if we again have a design change) :-).

Got that. I've removed those screenshots and changed text accordingly.
